context/src/env/parser.rs

```diff
@@ -64,6 +67,49 @@ pub enum CIPlatform {
     Drone,
 }
 
+#[cfg(feature = "ruby")]
+impl CIPlatform {
+    pub fn to_string(&self) -> &str {
+        match self {
+            CIPlatform::GitHubActions => "GITHUB_ACTIONS",
+            CIPlatform::JenkinsPipeline => "JENKINS_PIPELINE",
+            CIPlatform::CircleCI => "CIRCLECI",
+            CIPlatform::Buildkite => "BUILDKITE",
+            CIPlatform::Semaphore => "SEMAPHORE",
+            CIPlatform::TravisCI => "TRAVIS_CI",
+            CIPlatform::Webappio => "WEBAPPIO",
+            CIPlatform::AWSCodeBuild => "AWS_CODEBUILD",
+            CIPlatform::BitbucketPipelines => "BITBUCKET",
+            CIPlatform::AzurePipelines => "AZURE_PIPELINES",
+            CIPlatform::GitLabCI => "GITLAB_CI",
+            CIPlatform::Drone => "DRONE",
+        }
+    }
+}
+
+#[cfg(feature = "ruby")]
+impl magnus::TryConvert for CIPlatform {
+    fn try_convert(val: magnus::Value) -> Result<Self, magnus::Error> {
+        let val: i32 = val.funcall("to_i", ())?;
+        match val {
+            0 => Ok(CIPlatform::GitHubActions),
+            1 => Ok(CIPlatform::JenkinsPipeline),
+            2 => Ok(CIPlatform::CircleCI),
+            3 => Ok(CIPlatform::Buildkite),
+            4 => Ok(CIPlatform::Semaphore),
+            5 => Ok(CIPlatform::TravisCI),
+            6 => Ok(CIPlatform::Webappio),
+            7 => Ok(CIPlatform::AWSCodeBuild),
+            8 => Ok(CIPlatform::BitbucketPipelines),
+            9 => Ok(CIPlatform::AzurePipelines),
+            10 => Ok(CIPlatform::GitLabCI),
+            11 => Ok(CIPlatform::Drone),
+            // TODO
+            _ => Ok(CIPlatform::GitHubActions),
```

Comment:
Shouldn't this return `Err(magnus::Error)`?

Reply:
Nope, you can't construct Magnus errors without a ruby object. It should return an error, but it doesn't work with anyhow out of the box. The TODO here was to return an error, but I forget to add that in. Will add it.

Reply:
Nevermind, figured out how to get the Magnus error populated. You use `magnus::Ruby` directly and not the one populated in `Magnus::init`